executor/join.go

```diff
@@ -147,15 +141,9 @@ func (e *HashJoinExec) Open(ctx context.Context) error {
 	e.prepared = false
 	e.memTracker = memory.NewTracker(e.id, e.ctx.GetSessionVars().MemQuotaHashJoin)
 	e.memTracker.AttachTo(e.ctx.GetSessionVars().StmtCtx.MemTracker)
-
-	e.hashTableValBufs = make([][][]byte, e.concurrency)
-	e.hashJoinBuffers = make([]*hashJoinBuffer, 0, e.concurrency)
-	for i := uint(0); i < e.concurrency; i++ {
-		buffer := &hashJoinBuffer{
-			data:  make([]types.Datum, len(e.outerKeys)),
-			bytes: make([]byte, 0, 10000),
-		}
-		e.hashJoinBuffers = append(e.hashJoinBuffers, buffer)
+	e.joinKeyBuf = make([][]byte, e.concurrency)
+	for i := range e.joinKeyBuf {
+		e.joinKeyBuf[i] = make([]byte, 1)
```

Comment:
why setting the size to `1`?

Reply:
For memory reuse at https://github.com/pingcap/tidb/pull/11832/files#diff-671e79aa0fba47bd04f56fa4788120e2R360